packages/web/src/services/FileDownloadServiceWeb.ts

```diff
@@ -1,23 +1,113 @@
+import axios from "axios";
+import StreamSaver from "streamsaver";
+
 import {
     FileDownloadService,
     DownloadResult,
     FileInfo,
     DownloadResolution,
-    HttpServiceBase,
+    FileStorageServiceBase,
 } from "../../../core/services";
 
-export default class FileDownloadServiceWeb extends HttpServiceBase implements FileDownloadService {
+export default class FileDownloadServiceWeb
+    extends FileStorageServiceBase
+    implements FileDownloadService {
     isFileSystemAccessible = false;
 
     public async download(fileInfo: FileInfo): Promise<DownloadResult> {
+        if (fileInfo.path.endsWith(".zarr")) {
+            const downloadResult = await this.handleZarrFile(fileInfo);
+            if (downloadResult) {
+                return downloadResult;
+            }
+        }
+
+        return this.downloadFile(fileInfo);
+    }
+
+    private async handleZarrFile(fileInfo: FileInfo): Promise<DownloadResult | null> {
+        if (this.isS3Url(fileInfo.path)) {
+            return await this.downloadS3Directory(fileInfo);
+        }
+
+        const localDownloadResult = this.isLocalPath(fileInfo.path, fileInfo);
+        return localDownloadResult; // This will be either a DownloadResult or null
```

Comment:
Feels like we should throw an error here since we couldn't download it when null then?